plugins/modules/zypper.py

```diff
@@ -512,7 +512,8 @@ def repo_refresh(m):
 
 
 def transactional_updates():
-    return os.path.exists('/var/lib/misc/transactional-update.state')
+    return os.path.exists('/var/lib/misc/transactional-update.state') \
```

Comment:
```suggestion
    return any(map(os.path.exists, ['/var/lib/misc/transactional-update.state', '/usr/sbin/transactional-update']))
```
another way to write this

Reply:
I assume the suggested change makes the boolean operation non-trivial anymore, in the way that it allocates an array and has to reflectively apply the list items to the supplied function. For more items i'd be with you, but for two..?

Thx for the suggestion nevertheless! Phyton's not my native language and i'm happy to learn about new ways to do stuff :)

Reply:
On Python 3 it the `map` result is a generator, but on Python 2 this causes two calls to `os.path.exists`, while the original statement only makes two calls if the first one returns `False`.

But using a generator expression would make it rather efficient (though still slower than the original version) even on Python 2.6 and 2.7: `any(os.path.exists(path) for path in ('/var/lib/misc/transactional-update.state', '/usr/sbin/transactional-update'))`.

Reply:
Thank you for the explanation 🤗 Makes sense.

So should i change it, or leave it as is? What's preferred?

Reply:
Python 2 is dead and is probably irrelevant if something there is slower than the supported release. What does /var/lib/misc/transactional-update.state contain? And is the system in a valid state if this file is missing?

Reply:
On openSUSE microOS this file is missing - both as reported and as checked by me on a fresh install.
It's the proposed solution from the PR referenced above, and so far it seems to be a valid fix that works.

Reply:
I think I'm good here. Thank you for your work!

Reply:
Good point. I'll have to look into a "normal" openSUSE install to check if the transactional-update binary resides at the same place there and will  update the PR if it's the case.

Reply:
Both files are not present in openSUSE Tumbleweed and openSUSE Leap 15.4.

Reply:
> So should i change it, or leave it as is? What's preferred?

I would keep the current version.

> Python 2 is dead and is probably irrelevant if something there is slower than the supported release.

I wish it would really be dead, but unfortunately it is not. That said, I don't know what was the last OpenSuSE version where Python 2 was the default system Python, and whether that one is still in use. But for other OSes (Debian, RHEL) there are still a lot of old versions in production. (Recently someone noticed on ansible-project that current Ansible doesn't support Python 2.4 anymore, IIRC...)

Reply:
It's only present, when selecting the "Transactional Server" install method, which unses a read-only filesystem. 

But i just checked all three flavors on fresh installs: Tumbleweed, 15.4 and microOS and none contained the state-file. I'll change the PR accordingly to just check for the binary.

Reply:
Ping @sebix who implemented this in #3159

Reply:
Hi, thanks for pinging me.

Is the pure existence of the executable enough to check if transactional updates are actually in use? It may be true for MicroOS, which only exists in the transaction-variant. In openSUSE (e.g. Tumbleweed) I can as well install `transactional-update` and have the executable on the disk. That does not mean I actually activated transactional updates. That's the reason why I checked for the state file.

If there's a better way to check if transactional updates are really in use (and not only "possible"), I'm for the change. But not when we break the usage of the module on other systems.

Reply:
Is there a way to identify MicroOS? Maybe looking at `/etc/os-release` or `/etc/lsb-release`?

Reply:
@felixfontein in my MicroOS installation there is no ```/etc/lsb-release``` file, but ```/etc/os-release``` contains:
```
NAME="openSUSE MicroOS"
# VERSION="20230130"
ID="opensuse-microos"
ID_LIKE="suse opensuse opensuse-tumbleweed"
VERSION_ID="20230130"
PRETTY_NAME="openSUSE MicroOS"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:microos:20230130"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:MicroOS"
LOGO="distributor-logo-MicroOS"
```

Reply:
> Maybe looking at `/etc/os-release` or `/etc/lsb-release`?

It's id is `opensuse-microos`, so yes. 
But we shouldn't only rely on that, as the state-file doesn't exist on a transactional Tumbleweed/Leap install.

Reply:
I think we should replicate the checks zypper uses, if suitable. And it doesn't seem to use black magic, so we should be able to do so: https://github.com/openSUSE/zypper/blob/master/src/commands/conditions.cc#L44